README.md

```diff
@@ -1 +1,44 @@
-# MultiDID
+# multidid
+
+> A very compact representation of DIDs and DID URLs
+
+Multidid is a representation strategy for DIDs and DID URLs that is very compact and extensible. It allows any DID method to be represented as a string of bytes.
+
+## Format
+
+```
+<multidid-code><method-code><method-bytes><url-length><url-bytes>
+```
+
+Where:
+
+* `multidid-code` - the value `0x0d1d` encoded as a varint
+* `method-code` - a varint encoded multicode for the [DID Method](https://www.w3.org/TR/did-core/#a-simple-example)
+* `method-bytes` - data representing the method-id
```

Comment:
The method code identifies which DID method it is. The method bytes is the rest of the data for the method specific identifier. Lmk if you think this can be clarified!

Reply:
Oh it’s actual id, was just confused by the name (perhaps it should be called whatever did:core calls those instead)

Reply:
Also unless it’s length prefixed it would be impossible to tell where this fragment ends and next one starts

Reply:
> Also unless it’s length prefixed it would be impossible to tell where this fragment ends and next one starts

This is not the case. The `method-code` should tell the implementer how long the expected bytestring is. See Key DID for an example.

> Oh it’s actual id, was just confused by the name (perhaps it should be called whatever did:core calls those instead)

Yes, I think this makes sense. It's slightly nuanced though since in Key DID what we use as `method-code` is itself part of the "method-specific-identifier"

Reply:
I think what you mean is that `<method-code><method-bytes>` is it's own multiformat, meaning that `method-bytes` may contain additional varint tags in them when necessary e.g. did:web will have to length-prefix it because unlike did:key there's no way to infer the length.

I think it is worth mentioning this somehow in the spec, maybe even an example of did:web would be useful.

Reply:
I don't exactly know what you mean by "its own multiformat". Can you maybe make a suggestion with the edit you are thinking about?

Reply:
I think we might be in agreement here and just lack formal language to describe these things. In the comment above I proposed IPLD Schema extension syntax to describe this